config/v1/types_oauth.go

```diff
@@ -58,8 +58,10 @@ type TokenConfig struct {
 	// consecutive uses of the token. Tokens become invalid if they are not
 	// used within this temporal window. The user will need to acquire a new
 	// token to regain access once a token times out.
-	// If this value is not set, then tokens are valid until their expiry.
-	// Takes valid time duration string such as "5m", "1.5h" or "2h45m".
+	// Takes valid time duration string such as "5m", "-1s", "1.5h" or "2h45m".
+	// Duration >= 0s: Tokens time out if there is no activity for this duration. Allowed minimum value is `300s` (5 minutes).
+	// Duration = nil: Tokens time out is disabled (default).
+	// Duration < 0s: Tokens time out is enabled but tokens never timeout unless configured per client (e.g. `-1s`).
```

Comment:
strange API. Doesn't this qualify for another boolean option?

Where does the requirement come from to have the negative case in the first place?

Reply:
I think the weirdness comes from the time when the old APIs were being converted to the new ones directly, but for whatever reason, pointers were discouraged back then?

Since this is a pointer, this API does not make sense to be kept the same way

The doc of the original API:
```
// accessTokenInactivityTimeoutSeconds defines the default token
// inactivity timeout for tokens granted by any client.
// The value represents the maximum amount of time that can occur between
// consecutive uses of the token. Tokens become invalid if they are not
// used within this temporal window. The user will need to acquire a new
// token to regain access once a token times out.
// Valid values are integer values:
//   x < 0  Tokens time out is enabled but tokens never timeout unless configured per client (e.g. `-1`)
//   x = 0  Tokens time out is disabled (default)
//   x > 0  Tokens time out if there is no activity for x seconds
// The current minimum allowed value for X is 300 (5 minutes)

```

Reply:
Side-question: what does Duration == nil mean? Text says that timeout is enabled. Is it?